runtime/src/main/scala/com/soundcloud/twinagle/MessageFilter.scala

```diff
@@ -0,0 +1,11 @@
+package com.soundcloud.twinagle
+
+import com.twitter.finagle.SimpleFilter
+import scalapb.{GeneratedMessage, GeneratedMessageCompanion}
+
+trait MessageFilter {
+  def toFilter[
+      Req <: GeneratedMessage: GeneratedMessageCompanion,
+      Resp <: GeneratedMessage: GeneratedMessageCompanion
+  ]: SimpleFilter[Req, Resp]
```

Comment:
we don't need to restrict this to extending `SimpleFilter` (an abstract class)
```suggestion
  ]: Filter[Req, Resp, Req, Resp]
```